CHANGELOG.md

```diff
@@ -4,6 +4,10 @@
 
 - `[jest-snapshot]` Introduce `toMatchInlineSnapshot` and `toThrowErrorMatchingInlineSnapshot` matchers ([#6380](https://github.com/facebook/jest/pull/6380))
 
+### Fixes
+
+- `[jest-mock]` [**BREAKING**] Fix bugs with mock/spy result tracking of recursive functions ([#6381](https://github.com/facebook/jest/pull/6381)
```

Comment:
I wonder if we should avoid the breaking change now (so keep `isThrow` a boolean, etc) but fix the bug. The we can land the breaking change for Jest 24?

Last time we changed this (between two jest 23 betas) we broke FB's test suite.

FTR I agree this is better. But I'd like to avoid the breakage.

One solution is to add `type` instead of removing `isThrow`, and go with your original `undefined` approach. Then we could switch the internal matchers to use `type`, but not break compatibility.